### ISSUE TITLE
lifecycle hook doesn't give access to model relationships

### ISSUE THREAD

**Issue by xavier-rodet** [vuex-orm/vuex-orm#490]
Hi,

First of all, thank you for your awesome plugin, it is really simple & powerful!

I used to define **Accessors** to caculate dynamic properties dependending on model & relation values (like ratio etc ...) and it works fine !

Exemple : 
```js
  get averageStockPerCategory() {
    const totalStock = this.categories.reduce((total, category) => ({
      stock: total.stock+ category.stock
    })).stock;

    return totalStock / this.categories.length;
  }
```

However I would like to move this calculation from thoses _**Accessors**_ to **_afterCreate lifecycle hook_** in order to avoid calculations everytime i want to access thoses calculated properties. 

But it seems i cannot access relations from lifecycle hooks.

See that simple example case :
```js
export default class Parent extends Model {
// ...
  static afterCreate(model) {
   model.name; // does contain the value "parent"
   model.childs; // is empty array [], relation not accessible yet :(

  // I wish make my calculation here
  }
}
```

The way i'm fetching my model is by using a simple query : 

```js
this.api().get(apiUrl+ `/parent/${id}`);
```

My api is sending back a nested result like this : 
```json
{
	"id": 1,
	"name": "Parent",
	"childs": [
		{
			"id": 1,
			"name": "Child",
		}
	]
}
```

Thanks!

**Comment by kiaking** [comment]
Yes, unfortunately this is current limitation. You can't access relationships during the lifecycle hooks... When lazy loading is implemented, we'll be able to handle this one...

**Comment by xavier-rodet** [comment]
Good to know, thanks 👍
